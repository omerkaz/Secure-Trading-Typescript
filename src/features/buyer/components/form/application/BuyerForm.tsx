import { Field, Form, Formik } from 'formik';
import { Dispatch, SetStateAction } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import * as Yup from 'yup';

import Button from '@components/button/Button';
import FormElement from '@components/form/FormElement';

import { generateSmsCode } from '@utils/codeGenerator/generateCode';
import { buyerFormInitialValues as initialValues } from '@utils/formsInitialValue/initialValues';
import { buyerValidationSchema } from '@utils/yup/buyerValidationSchema';

import {
    createBuyerApplication,
    getReferenceCode,
} from '../../../buyerApplicationSlice';
import SellerInformation from './SellerInformation';

interface IProps {
    nextForm: Dispatch<SetStateAction<number>>;
}
type TBuyer = Yup.InferType<typeof buyerValidationSchema>;

function BuyerForm({ nextForm }: IProps) {
    const dispatch = useDispatch();
    // take reference code in matched seller for put in buyer ınformation object
    const sellerReferenceCode = useSelector(getReferenceCode);
    return (
        <div>
            <p className="text-center fs-4 mb-0">Satıcı Bilgileri</p>
            <div className="d-flex justify-content-center border">
                <SellerInformation />
            </div>
            <p className="text-center fs-4 mb-0">Alıcı Bilgileri</p>
            <Formik
                initialValues={initialValues}
                onSubmit={(values: TBuyer) => {
                    dispatch(
                        createBuyerApplication({
                            values,
                            SMSCode: generateSmsCode(),
                            referenceCode: sellerReferenceCode,
                        }),
                    );
                    nextForm((prevValue) => prevValue + 1);
                }}
                validationSchema={buyerValidationSchema}
            >
                {({ values, errors, handleChange, touched }) => (
                    <Form>
                        <div className="border rounded px-3 pt-2 ">
                            <div className="row mb-md-2 mb-sm-1 ">
                                <div className="form-outline d-flex justify-content-center">
                                    <Field
                                        className="w-50 border rounded"
                                        as="select"
                                        name="legalOrRealPerson"
                                        values={values.legalOrRealPerson}
                                        style={{ height: '3.5vmin' }}
                                    >
                                        <option value="gerçek">
                                            Gerçek Kişi
                                        </option>
                                        <option value="tüzel">
                                            Tüzel Kişi
                                        </option>
                                    </Field>
                                </div>
                                <div className="form-outline mt-2 d-flex justify-content-center">
                                    <FormElement
                                        as={'input'}
                                        placeholder={'Tc Kimlik Numarası'}
                                        name={'TCNumber'}
                                        values={values.TCNumber}
                                        errors={errors.TCNumber}
                                        touched={touched.TCNumber}
                                        handleChange={handleChange}
                                    />
                                </div>
                                <div className="form-outline mt-2 d-flex justify-content-center">
                                    <FormElement
                                        as={'input'}
                                        placeholder={'İsim Soyisim'}
                                        name={'fullName'}
                                        values={values.fullName}
                                        errors={errors.fullName}
                                        touched={touched.fullName}
                                        handleChange={handleChange}
                                    />
                                </div>
                                <div className="form-outline my-2 d-flex justify-content-center">
                                    <FormElement
                                        as={'input'}
                                        placeholder={'Alıcı Telefon Numarası'}
                                        name={'buyerPhoneNumber'}
                                        values={values.buyerPhoneNumber}
                                        errors={errors.buyerPhoneNumber}
                                        touched={touched.buyerPhoneNumber}
                                        handleChange={handleChange}
                                    />
                                </div>
                                <div className="d-flex justify-content-center mt-2">
                                    <Button children={'Gönder'}></Button>
                                </div>
                            </div>
                        </div>
                    </Form>
                )}
            </Formik>
        </div>
    );
}

export default BuyerForm;
