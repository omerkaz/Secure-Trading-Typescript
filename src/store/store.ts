import { configureStore } from '@reduxjs/toolkit';
import sellerApplicationSlice from '@features/seller/sellerApplicationSlice';
import buyerApplicationSlice from '@features/buyer/buyerApplicationSlice';
import applicationStatusSlice from '@features/status/applicationStatusSlice';
import applicationCancelSlice from '@features/cancel/applicationCancelSlice';
const store = configureStore({
    reducer: {
        sellers: sellerApplicationSlice,
        buyers: buyerApplicationSlice,
        status: applicationStatusSlice,
        cancel:applicationCancelSlice,
    },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch


export default store;
