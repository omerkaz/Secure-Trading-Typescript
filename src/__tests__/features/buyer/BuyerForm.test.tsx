import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from 'store/store';

import BuyerForm from '@features/buyer/components/form/application/BuyerForm';

describe('BuyerForm elements renders correctly', () => {
    test('snapshot test', () => {
        const nextForm = jest.fn();
        const componentTree = render(
            <Provider store={store}>
                <BuyerForm nextForm={nextForm} />
            </Provider>,
        );
        expect(componentTree).toMatchSnapshot();
    });

    test('elements test', () => {
        const nextForm = jest.fn();
        render(
            <Provider store={store}>
                <BuyerForm nextForm={nextForm} />
            </Provider>,
        );
        const sellerHeader = screen.getByText('Sat覺c覺 Bilgileri');
        expect(sellerHeader).toBeInTheDocument();
        const sellersCardElement = screen.getByText('Referans No:');
        expect(sellersCardElement).toBeInTheDocument();
        const buyerHeader = screen.getByText('Al覺c覺 Bilgileri');
        expect(buyerHeader).toBeInTheDocument();
    });
});
