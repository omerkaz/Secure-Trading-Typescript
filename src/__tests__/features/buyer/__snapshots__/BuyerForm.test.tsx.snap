// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyerForm elements renders correctly snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <p
          class="text-center fs-4 mb-0"
        >
          Satıcı Bilgileri
        </p>
        <div
          class="d-flex justify-content-center border"
        >
          <div
            class="card border-0"
            style="width: 30vw;"
          >
            <div
              class="card-body"
            >
              <p
                class="card-text"
              >
                Referans No:
              </p>
              <p
                class="card-text"
              >
                Adı Soyadı: 
              </p>
              <p
                class="card-text"
              >
                Cep Telefonu: 
              </p>
              <p
                class="card-text"
              >
                Araç Plakası: 
              </p>
              <p
                class="card-text"
              >
                Satış Tutarı: 
              </p>
            </div>
          </div>
        </div>
        <p
          class="text-center fs-4 mb-0"
        >
          Alıcı Bilgileri
        </p>
        <form
          action="#"
        >
          <div
            class="border rounded px-3 pt-2 "
          >
            <div
              class="row mb-md-2 mb-sm-1 "
            >
              <div
                class="form-outline d-flex justify-content-center"
              >
                <select
                  class="w-50 border rounded"
                  name="legalOrRealPerson"
                  values="gerçek"
                >
                  <option
                    value="gerçek"
                  >
                    Gerçek Kişi
                  </option>
                  <option
                    value="tüzel"
                  >
                    Tüzel Kişi
                  </option>
                </select>
              </div>
              <div
                class="form-outline mt-2 d-flex justify-content-center"
              >
                <div
                  class="col-md-6 col-sm-12 mt-2"
                >
                  <div
                    class="form-outline"
                  >
                    <label
                      class="d-md-none d-sm-block text-center mb-1"
                    >
                      Tc Kimlik Numarası
                    </label>
                    <input
                      class="w-100 border rounded"
                      name="TCNumber"
                      placeholder="Tc Kimlik Numarası"
                      value=""
                      values=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="form-outline mt-2 d-flex justify-content-center"
              >
                <div
                  class="col-md-6 col-sm-12 mt-2"
                >
                  <div
                    class="form-outline"
                  >
                    <label
                      class="d-md-none d-sm-block text-center mb-1"
                    >
                      İsim Soyisim
                    </label>
                    <input
                      class="w-100 border rounded"
                      name="fullName"
                      placeholder="İsim Soyisim"
                      value=""
                      values=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="form-outline my-2 d-flex justify-content-center"
              >
                <div
                  class="col-md-6 col-sm-12 mt-2"
                >
                  <div
                    class="form-outline"
                  >
                    <label
                      class="d-md-none d-sm-block text-center mb-1"
                    >
                      Alıcı Telefon Numarası
                    </label>
                    <input
                      class="w-100 border rounded"
                      name="buyerPhoneNumber"
                      placeholder="Alıcı Telefon Numarası"
                      value=""
                      values=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="d-flex justify-content-center mt-2"
              >
                <button
                  class="button"
                  type="submit"
                >
                  <span
                    style="color: rgb(221, 221, 221);"
                  >
                    Gönder
                  </span>
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <p
        class="text-center fs-4 mb-0"
      >
        Satıcı Bilgileri
      </p>
      <div
        class="d-flex justify-content-center border"
      >
        <div
          class="card border-0"
          style="width: 30vw;"
        >
          <div
            class="card-body"
          >
            <p
              class="card-text"
            >
              Referans No:
            </p>
            <p
              class="card-text"
            >
              Adı Soyadı: 
            </p>
            <p
              class="card-text"
            >
              Cep Telefonu: 
            </p>
            <p
              class="card-text"
            >
              Araç Plakası: 
            </p>
            <p
              class="card-text"
            >
              Satış Tutarı: 
            </p>
          </div>
        </div>
      </div>
      <p
        class="text-center fs-4 mb-0"
      >
        Alıcı Bilgileri
      </p>
      <form
        action="#"
      >
        <div
          class="border rounded px-3 pt-2 "
        >
          <div
            class="row mb-md-2 mb-sm-1 "
          >
            <div
              class="form-outline d-flex justify-content-center"
            >
              <select
                class="w-50 border rounded"
                name="legalOrRealPerson"
                values="gerçek"
              >
                <option
                  value="gerçek"
                >
                  Gerçek Kişi
                </option>
                <option
                  value="tüzel"
                >
                  Tüzel Kişi
                </option>
              </select>
            </div>
            <div
              class="form-outline mt-2 d-flex justify-content-center"
            >
              <div
                class="col-md-6 col-sm-12 mt-2"
              >
                <div
                  class="form-outline"
                >
                  <label
                    class="d-md-none d-sm-block text-center mb-1"
                  >
                    Tc Kimlik Numarası
                  </label>
                  <input
                    class="w-100 border rounded"
                    name="TCNumber"
                    placeholder="Tc Kimlik Numarası"
                    value=""
                    values=""
                  />
                </div>
              </div>
            </div>
            <div
              class="form-outline mt-2 d-flex justify-content-center"
            >
              <div
                class="col-md-6 col-sm-12 mt-2"
              >
                <div
                  class="form-outline"
                >
                  <label
                    class="d-md-none d-sm-block text-center mb-1"
                  >
                    İsim Soyisim
                  </label>
                  <input
                    class="w-100 border rounded"
                    name="fullName"
                    placeholder="İsim Soyisim"
                    value=""
                    values=""
                  />
                </div>
              </div>
            </div>
            <div
              class="form-outline my-2 d-flex justify-content-center"
            >
              <div
                class="col-md-6 col-sm-12 mt-2"
              >
                <div
                  class="form-outline"
                >
                  <label
                    class="d-md-none d-sm-block text-center mb-1"
                  >
                    Alıcı Telefon Numarası
                  </label>
                  <input
                    class="w-100 border rounded"
                    name="buyerPhoneNumber"
                    placeholder="Alıcı Telefon Numarası"
                    value=""
                    values=""
                  />
                </div>
              </div>
            </div>
            <div
              class="d-flex justify-content-center mt-2"
            >
              <button
                class="button"
                type="submit"
              >
                <span
                  style="color: rgb(221, 221, 221);"
                >
                  Gönder
                </span>
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
